src/main/java/org/radarcns/webapp/resource/SensorEndPoint.java

```diff
@@ -0,0 +1,246 @@
+    @ApiResponse(responseCode = "200", description =
+            "Returns a dataset.avsc object containing last "
+                    + "computed sample for the given inputs formatted either Acceleration.avsc or "
+                    + "DoubleValue.avsc")
+    @ApiResponse(responseCode = "401", description = "Access denied error occurred")
+    @ApiResponse(responseCode = "403", description = "Not Authorised error occurred")
+    @NeedsPermissionOnSubject(entity = MEASUREMENT, operation = READ)
+    public Dataset getLastReceivedSampleJson(
+            @Alphanumeric @PathParam(SENSOR) String sensor,
+            @PathParam(STAT) DescriptiveStatistic stat,
+            @PathParam(INTERVAL) TimeWindow interval,
+            @Alphanumeric @PathParam(PROJECT_NAME) String projectName,
+            @Alphanumeric @PathParam(SUBJECT_ID) String subjectId,
+            @Alphanumeric @PathParam(SOURCE_ID) String sourceId) throws IOException {
+        mpClient.getSubject(subjectId);
+        if (SubjectDataAccessObject.exist(subjectId, mongoClient)) {
+            Dataset dataset = SensorDataAccessObject.getInstance()
+                    .getLastReceivedSample(subjectId, sourceId,
+                            stat, interval, sensor, mongoClient);
+
+            if (dataset.getDataset().isEmpty()) {
+                LOGGER.debug("No data for the subject {} with source {}", subjectId, sourceId);
+            }
+
+            return dataset;
+        } else {
+            return emptyDataset(subjectId, sourceId, sensor, stat, interval,
+                    Instant.now());
+
+        }
+    }
+
+    //--------------------------------------------------------------------------------------------//
+    //                                   WHOLE-DATA FUNCTIONS                                     //
+    //--------------------------------------------------------------------------------------------//
+
+    /**
+     * JSON function that returns all available samples for the given data.
+     */
+    @GET
+    @Produces({APPLICATION_JSON, AVRO_BINARY})
+    @Path("/{" + SENSOR + "}/{" + STAT + "}/{" + INTERVAL + "}/{" + PROJECT_NAME + "}/{"
+            + SUBJECT_ID + "}/{"
+            + SOURCE_ID + "}")
+    @Operation(summary = "Returns a dataset object formatted in JSON.",
+            description = "Each collected sample is aggregated to provide near real-time "
+                    + "statistical "
+                    + "results. This end-point returns all available results of type stat for the "
+                    + "given subjectID, sourceID, and sensor. Data can be queried using different "
+                    + "time-frame resolutions. The response is formatted in JSON.")
+    @ApiResponse(responseCode = "500", description = "An error occurs while executing, in the body "
+            + "there is a message.avsc object with more details.")
+    @ApiResponse(responseCode = "204", description =
```

Comment:
We should update the javadocs to say 404 too

Reply:
Done